### PR TITLE
🎁 prepare release 0.6.0

### PR DESCRIPTION
**Pull request by JakeChampion**
Note: I've bumped from 0.5.1 to 0.6.0 due to upgrading wasmtime to a new major version (10.0.0)

Cutting a new release, this currently contains steps 1 through to 5 of [Releasing Viceroy](https://github.com/fastly/Viceroy/blob/main/doc/RELEASING.md#releasing-viceroy).

> Below are the steps needed to do a Viceroy release:
> 
> 1. Make sure the Viceroy version has been bumped up to the current release
>    version. You might need to bump the minor version (e.g. 0.2.0 to 0.3.0) if
>    there are any semver breaking changes. Review the changes since the last
>    release just to be sure.
> 1. Update the `Cargo.lock` files by running `make generate-lockfile`.
> 1. Update `CHANGELOG.md` so that it contains all of the updates since the
>    previous version as its own commit.
> 1. Create a local branch in the form `release-x.y.z` where `x`, `y`, and `z` are
>    the major, minor, and patch versions of Viceroy and have the tip of the
>    branch contain the Changelog commit.
> 1. Run `make ci` locally to make sure that everything will pass before pushing
>    the branch and opening up a PR.
> 1. After you get approval, run `git tag vx.y.z HEAD && git push origin --tags`.
>    Pushing this tag will kick off a build for all of the release artifacts.
> 1. After CI completes, we should publish each crate in the workspace to the
>    crates.io registry. Note that we must do this in order of dependencies. So,
>   1. `(cd lib && cargo publish)`
>   1. `(cd cli && cargo publish)`
> 1. Now, we should return to our release PR.
>   1. Update the version fields in `lib/Cargo.toml` and `cli/Cargo.toml` to the
>      next patch version (so `z + 1`).
>   1. Update the dependency on `viceroy-lib` in `cli/Cargo.toml` to the next
>      patch version.
>   1. Update all the lockfiles by running `make generate-lockfile`.
>   1. Restore the `## Unreleased` header at the top of `CHANGELOG.md`.
> 1. Get another approval and merge when CI passes.
>

### CODE REVIEW

**Comment by acw** [comment]
This looks fine, but the fact that the Ubuntu CI job is failing is worrying. Any ideas? (If this is a known flake in our CI, I can approve, but I'm hesitant to hit approve with a CI job failing.)

**Review by acw — APPROVED** [review]
LGTM!

**Review by acfoltzer — DISMISSED** [review]
Approved for release pending CI :shipit:

**Comment by JakeChampion** [comment]
@acfoltzer the release has happened, this is now ready for final reviews before merging

**Review by acfoltzer — APPROVED** [review]

**Pull request merged**
